app/models/comment.rb

```diff
@@ -0,0 +1,13 @@
+class Comment < ActiveRecord::Base
+
+  has_many :comments, as: :commentable
+
+  belongs_to :commentable, polymorphic: true
+  belongs_to :user
+
+  validates :content, presence: true
+  validates :user, presence: true
```

Comment:
Cool thanks, and yes I ought to be validating that as well. After what you said yesterday about validations not needing to be done on optional attributes, it certainly makes sense.